### ISSUE TITLE
`_fit_minimize` requires update of `no_hess` list: L-BFGS-B and TNC do not use Hessian information (hess)

### ISSUE THREAD

**Issue by pcuestas** [statsmodels/statsmodels#9140]
L-BFGS-B and TNC do not use Hessian information (hess). Therefore, these methods should be added to 

https://github.com/statsmodels/statsmodels/blob/23faea30e30ff759c3c9d3f1d57864b2aa68c827/statsmodels/base/optimizer.py#L341

**Comment by josef-pkt** [comment]
yes lbfgs are missing but AFAICS we don't have a wrapper for fmin_tnc.

Do you get any problems, exception, ... when running the current code?
I don't understand why this doesn't cause problems when we run the unit tests.

**update**
`no_hess` is only used in `_fit_minimize` and we have few unit tests for it.

**Comment by josef-pkt** [comment]
we should get at least a smoke test for all methods in `_fit_minimize`.

**Comment by pcuestas** [comment]
I do not have any problem when running the current code, only a warning. Sorry, I should have mentioned this. I just wanted to point this out, as I believe it can be confusing for users to get this warning when there shouldn't be one.
